### PR TITLE
Add cloudprovider support to rke2

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/rke2/issues/142

### CODE REVIEW

File: pkg/cli/cmds/root.go [hunk 1]
```diff
@@ -25,6 +25,18 @@ var (
 			EnvVars:     []string{"RKE2_REPO"},
 			Destination: &config.Repo,
 		},
+		&cli.StringFlag{
```
Comment (briandowns) [inline pkg/cli/cmds/root.go:28]: Are these dependent on one another? Should we have a check to see if both are present and not just one?

Comment (galal-hussein) [reply pkg/cli/cmds/root.go:28]: well it depends, so cloud config doesnt work without cloud provider name, but the opposite is not true, so you can pass only cloud provider name  for some providers without the need to set up the cloud config file

Comment (galal-hussein) [reply pkg/cli/cmds/root.go:28]: I will add a check if the config is supported without the name

File: pkg/rke2/server.go [hunk 2]
```diff
@@ -74,11 +77,23 @@ func setup(ctx *cli.Context, cfg Config) error {
 
 	managed.RegisterDriver(&etcd.ETCD{})
 
+	var cpConifg *podexecutor.CloudProviderConfig
```
Comment (superseb) [inline pkg/rke2/server.go:80]: typo

Comment (galal-hussein) [reply pkg/rke2/server.go:80]: Thanks for spotting that, will fix

File: pkg/rke2/server.go [hunk 3]
```diff
@@ -74,11 +77,23 @@ func setup(ctx *cli.Context, cfg Config) error {
 
 	managed.RegisterDriver(&etcd.ETCD{})
 
+	var cpConfig *podexecutor.CloudProviderConfig
+	if cfg.CloudProviderConfig != "" && cfg.CloudProviderName == "" {
+		return fmt.Errorf("cloud provider name has to be provided with cloud config path")
```
Comment (brandond) [inline pkg/rke2/server.go:82]: The message here could be confusing, maybe: "--cloud-provider-config requires --cloud-provider-name"

**Pull request merged**